src/http/validations/device.js

```diff
@@ -13,6 +14,19 @@ const getSingleDevice = (req, res, next) => {
   return next();
 };
 
-export default {
-  getSingleDevice,
+const updateDevice = (req, res, next) => {
+  const { deviceId } = req.params;
+
+  const fieldErrors = new FieldErrors();
+
+  if (!db.validResourceId(deviceId)) fieldErrors.addError('deviceId', 'deviceId should be a ');
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is a required string');
```

Comment:
DeviceId coming from the request params would always be a string so there is no need for this validation.

Reply:
fixed